pkg/platform/kube/test/platform_test.go

```diff
@@ -763,6 +764,93 @@ func (suite *DeployFunctionTestSuite) TestCreateFunctionWithTemplatedIngress() {
 		})
 }
 
+func (suite *DeployFunctionTestSuite) TestFunctionSecretCreation() {
+
+	// set platform config to support scrubbing
+	suite.PlatformConfiguration.SensitiveFields.MaskSensitiveFields = true
+
+	// reset platform configuration when done
+	defer func() {
+		suite.PlatformConfiguration.SensitiveFields.MaskSensitiveFields = false
+	}()
+
+	functionName := "func-with-secret"
+	password := "1234"
+	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
+	secretName := fmt.Sprintf("%s%s", functionconfig.NuclioSecretNamePrefix, functionName)
+
+	createFunctionOptions.FunctionConfig.Spec.Build.CodeEntryAttributes = map[string]interface{}{
+		"password": password,
+	}
+
+	// delete function secrets when done
+	defer func() {
+		secrets, err := suite.KubeClientSet.CoreV1().Secrets(suite.Namespace).List(suite.Ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", common.NuclioResourceLabelKeyFunctionName, functionName),
+		})
+		if err != nil {
+			suite.Logger.WarnWith("Failed to list secrets", "err", err)
+		}
+
+		for _, secret := range secrets.Items {
+			err := suite.KubeClientSet.CoreV1().Secrets(suite.Namespace).Delete(suite.Ctx, secret.Name, metav1.DeleteOptions{})
+			if err != nil {
+				suite.Logger.WarnWith("Failed to delete secret", "err", err, "secretName", secret.Name)
+			}
+		}
+	}()
+
+	// deploy function
+	suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+
+		// get function secrets
+		secrets, err := suite.KubeClientSet.CoreV1().Secrets(suite.Namespace).List(suite.Ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", common.NuclioResourceLabelKeyFunctionName, functionName),
+		})
```

Comment:
id move to kube.platform as a functionality
given function name, function namespace - get me its secrets